AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario1Tests.swift

```diff
@@ -190,49 +179,32 @@ class GraphQLConnectionScenario1Tests: XCTestCase {
         XCTAssertEqual(resultsArray.count, 2)
     }
 
-    func createTeam(id: String = UUID().uuidString, name: String) -> Team1? {
+    func createTeam(id: String = UUID().uuidString, name: String) async throws ->  Team1? {
         let team = Team1(id: id, name: name)
         var result: Team1?
-        let requestInvokedSuccessfully = expectation(description: "request completed")
-        Amplify.API.mutate(request: .create(team)) { event in
+        let event = try await Amplify.API.mutate(request: .create(team))
             switch event {
-            case .success(let data):
-                switch data {
                 case .success(let team):
                     result = team
-                default:
-                    XCTFail("Could not get data back")
-                }
-                requestInvokedSuccessfully.fulfill()
-            case .failure(let error):
-                XCTFail("Failed \(error)")
+            case .failure(let graphQLResponseError):
+                    XCTFail("Failed with error: \(graphQLResponseError)")
             }
-        }
-        wait(for: [requestInvokedSuccessfully], timeout: TestCommonConstants.networkTimeout)
         return result
     }
 
     func createProject(id: String = UUID().uuidString,
-                       name: String? = nil,
-                       team: Team1? = nil) -> Project1? {
+                        name: String? = nil,
+                        teamID: String,
+                        team: Team1? = nil) async throws -> Project1? {
```

Comment:
```suggestion
                        team: Team1? = nil) async throws -> Project1 {
```
Return `Project1` not optional since that's the return type of the inner switch statement. The goal is to reduce the nested result types by throwing the result's failure case. the method will have this structure:
```
func methodThatThrowsTwoDifferentErrors() async throws -> Project1 {
  let result = try await methodThatThrowsAndReturnsResult()
  switch result {
   case .success(let project): return project
   case .failure(let error): throw error
  }
}
```

Reply:
my mistake, it the return type should still be optional `Project1` if the `case .success(let project)` provides `project` as an optional type

Reply:
done! got you resolving the con